src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -57,13 +132,152 @@ public boolean equals(Object other) {
         }
 
         DeleteCommand otherDeleteCommand = (DeleteCommand) other;
-        return targetIndex.equals(otherDeleteCommand.targetIndex);
+
+        if (name != null && !name.equals(otherDeleteCommand.name)) {
+            return false;
+        }
+
+        if (nric != null && !nric.equals(otherDeleteCommand.nric)) {
+            return false;
+        }
+
```

Comment:
This seems like you wanted to check if `name` and `nric` are equivalent? `Objects.equal` might be a better choice here.